massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -919,6 +940,555 @@ fn cancel_async_message() {
+            // so the user should be refunded
+            match changes
+                .ledger_changes
+                .0
+                .get(
+                    &Address::from_str("AU1TyzwHarZMQSVJgxku8co7xjrRLnH74nFbNpoqNd98YhJkWgi")
+                        .unwrap(),
+                )
+                .unwrap()
+            {
+                SetUpdateOrDelete::Update(change_sc_update) => {
+                    assert_eq!(
+                        change_sc_update.balance,
+                        SetOrKeep::Set(Amount::from_str("110").unwrap())
+                    );
+                }
+                _ => panic!("wrong change type"),
+            }
+
+            assert_eq!(changes.deferred_call_changes.slots_change.len(), 2);
+            let (_slot, slot_change) = changes
+                .deferred_call_changes
+                .slots_change
+                .first_key_value()
+                .unwrap();
+
+            let (_id, set_delete) = slot_change.calls.first_key_value().unwrap();
+
+            // call was executed and then deleted
+            assert_eq!(set_delete, &SetOrDelete::Delete);
+
+            // assert total gas was set to 0
+            assert_eq!(changes.deferred_call_changes.total_gas, SetOrKeep::Set(0));
+            finalized_waitpoint_trigger_handle2.trigger();
+        });
+
+    let registry = DeferredCallRegistry::new(foreign_controllers.db.clone());
+
+    let mut defer_reg_slot_changes = DeferredRegistrySlotChanges {
+        calls: BTreeMap::new(),
+        gas: massa_deferred_calls::DeferredRegistryGasChange::Keep,
+        base_fee: massa_deferred_calls::DeferredRegistryBaseFeeChange::Keep,
+    };
+
+    defer_reg_slot_changes.set_base_fee(Amount::from_str("0.000005").unwrap());
+
+    let mut slot_changes = BTreeMap::default();
+    slot_changes.insert(
+        Slot {
+            period: 1,
+            thread: 1,
+        },
+        defer_reg_slot_changes,
+    );
+
+    let mut db_batch = DBBatch::default();
+
+    registry.apply_changes_to_batch(
+        DeferredRegistryChanges {
+            slots_change: slot_changes,
+            total_gas: SetOrKeep::Keep,
+        },
+        &mut db_batch,
+    );
+
+    foreign_controllers
+        .db
+        .write()
+        .write_batch(db_batch, DBBatch::default(), Some(Slot::new(1, 0)));
+    final_state_boilerplate(
+        &mut foreign_controllers.final_state,
+        foreign_controllers.db.clone(),
+        &foreign_controllers.selector_controller,
+        &mut foreign_controllers.ledger_controller,
+        Some(saved_bytecode),
+        None,
+        None,
+        Some(DeferredCallRegistry::new(foreign_controllers.db.clone())),
+    );
+
+    let mut universe = ExecutionTestUniverse::new(foreign_controllers, exec_cfg);
+
+    // abi call to register a deferred call
+    universe.deploy_bytecode_block(
+        &keypair,
+        Slot::new(1, 0),
+        include_bytes!("./wasm/deferred_call_register.wasm"),
```

Comment:
don't forget to put the sources of your new SCs in a PR there: https://github.com/massalabs/massa-unit-tests-src

Reply:
done https://github.com/massalabs/massa-unit-tests-src/pull/42